### PR TITLE
Add Telegram support

### PR DESCRIPTION
**Pull request by sofiia-hlyniana**
basic telegram support

### CODE REVIEW

**Comment by ilyabirman** [comment]
Thanks! Have you installed it anywhere already? I’d like to try it first.

Also, could you please respect the paddings in likely.styl and use lower case for hex digits?

File: source/services/index.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ var services = {
     pinterest:     require('./pinterest'),
     facebook:      require('./facebook'),
     twitter:       require('./twitter'),
-    gplus:         require('./gplus')
+    gplus:         require('./gplus'),
+    telegram:      require('./telegram')
```
Comment (NickMitin) [inline source/services/index.js]: Please, put comma in the end of last line. Commits will be cleaner.

File: source/svg.json [hunk 2]
```diff
@@ -4,5 +4,6 @@
...
+    "odnoklassniki": "8 6.107c.888 0 1.607-.72 1.607-1.607 0-.888-.72-1.607-1.607-1.607s-1.607.72-1.607 1.607c0 .888.72 1.607 1.607 1.607zM13 0H3C1 0 0 1 0 3v10c0 2 1 3 3 3h10c2 0 3-1 3-3V3c0-2-1-3-3-3zM8 .75c2.07 0 3.75 1.68 3.75 3.75 0 2.07-1.68 3.75-3.75 3.75S4.25 6.57 4.25 4.5C4.25 2.43 5.93.75 8 .75zm3.826 12.634c.42.42.42 1.097 0 1.515-.21.208-.483.313-.758.313-.274 0-.548-.105-.758-.314L8 12.59 5.69 14.9c-.42.418-1.098.418-1.516 0s-.42-1.098 0-1.516L6.357 11.2c-1.303-.386-2.288-1.073-2.337-1.11-.473-.354-.57-1.025-.214-1.5.354-.47 1.022-.567 1.496-.216.03.022 1.4.946 2.698.946 1.31 0 2.682-.934 2.693-.943.474-.355 1.146-.258 1.5.213.355.474.26 1.146-.214 1.5-.05.036-1.035.723-2.338 1.11l2.184 2.184",
+    "telegram": "6.4 10.9l5.6 4.7c.6.4 1.1.2 1.3-.7l2.3-12.2c.2-1.1-.4-1.5-1-1.2l-13.5 5.8c-.9.4-.9 1-.2 1.3l3.5 1.2 8-5.7c.4-.3.7-.1.4.2m-6.7 10.6c-.5 0-.4-.2-.5-.7l-1.3-5 10.2-6.9m-8.4 12.6c-.5 0-.4-.2-.5-.7l-1.3-5 10.2-6.9"
```
Comment (NickMitin) [inline source/svg.json]: Please, put comma in the end of last line. Commits will be cleaner.

File: styles/likely.styl [hunk 3]
```diff
@@ -160,6 +160,7 @@ colorize('gplus',         rgb(221,66,65)); // #dd4241
 colorize('vkontakte',     rgb(82,110,143)); // #526e8f);
 colorize('pinterest',     rgb(204,0,43)) // #cc002b
 colorize('odnoklassniki', #f6903b);
+colorize('telegram', #34ABDE)
```
Comment (NickMitin) [inline styles/likely.styl]: Inconsistency (no semicolon).

File: styles/likely.styl [hunk 4]
```diff
@@ -160,6 +160,7 @@ colorize('gplus',         rgb(221,66,65)); // #dd4241
 colorize('vkontakte',     rgb(82,110,143)); // #526e8f);
 colorize('pinterest',     rgb(204,0,43)) // #cc002b
```
Comment (NickMitin) [inline styles/likely.styl]: Inconsistency (no semicolon).

File: source/services/index.js (hunk 1 above)
Comment (iamakulov) [inline source/services/index.js]: @NickMitin We don’t follow this rule at the moment, so this shouldn’t be done yet

File: source/svg.json (hunk 2 above)
Comment (iamakulov) [inline source/svg.json]: @NickMitin As far as I know JSON doesn’t support trailing commas.

File: styles/likely.styl (hunk 3 above)
Comment (iamakulov) [inline styles/likely.styl]: Also, `#34ABDE` should be in the lower case and aligned with other color values, such as:

```js
colorize('odnoklassniki', #f6903b);  
colorize('telegram',      #34abde) ;
```

**Comment by iamakulov** [comment]
@ilyabirman, here’s now the button looks like:

[](http://i.imgur.com/HVsWFY6.png)

**Comment by iamakulov** [comment]
@sofiia-hlyniana, thank you, the button works perfectly! @NickMitin and I added some nits about the code. Could you please fix them? I’d be happy to merge the PR when they are done.

File: source/services/index.js (hunk 1 above)
Comment (vitkarpov) [inline source/services/index.js]: Seems it's time to add `eslint` ;)

Comment (iamakulov) [inline source/services/index.js]: @vitkarpov, sure, have already thought about it. If you’d like to setup it based on the current styling and make a PR, feel free to do so :-)

Comment (vitkarpov) [inline source/services/index.js]: https://github.com/ilyabirman/Likely/issues/28

**Comment by iamakulov** [comment]
@sofiia-hlyniana, I’ve forgotten about one more point. Updating the release files in a PR is an undesirable thing because it could cause conflicts with other PRs. Could you please revert the changes in these files?

**Comment by iamakulov** [comment]
@ilyabirman, are you OK with the Telegram button style? I’ve made required code changes locally, and I’m ready to merge this.

**Comment by ilyabirman** [comment]
The airplane should have some extra transparent lines inside :-) Will make tomorrow, if nobody else does.

**Comment by ilyabirman** [comment]
xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:a="http://ns.adobe.com/AdobeSVGViewerExtensions/3.0/"
	 x="0px" y="0px" width="16px" height="14.01px" viewBox="0 0 16 14.01" enable-background="new 0 0 16 14.01" xml:space="preserve"
	>
<defs>
</defs>
<g>
	<g>
		<path d="M6,11L5,8l11-8L0.622,5.914c0,0-0.672,0.23-0.619,0.655c0.053,0.425,0.602,0.619,0.602,0.619l3.575,1.203L5.8,13.545
			l2.742-2.411l-0.007-0.005l3.607,2.766c0.973,0.425,1.327-0.46,1.327-0.46L16,0L6,11z"/>
	</g>
</g>
</svg>

**Comment by iamakulov** [comment]
Thanks!

The updated button:

[](http://i.imgur.com/zTf3kHO.png)

**Comment by ilyabirman** [comment]
Could you please put the Telegram logo 1px lower? :-)

**Comment by iamakulov** [comment]
@ilyabirman, sure. Could you please do this on the icon level, i. e. by moving the icon 1px down on the canvas? The icon is 16×14.1, and the canvas is 16×16, so there should be enough place. There’s no quick non-hacky way to do this in another place, unfortunately.

**Comment by iamakulov** [comment]
@ilyabirman, nevermind, found the appropriate editor and moved the icon on my own.

Does this look well?

screencapture-file-d-ivan-likely-test-test-html-1462869225166

**Comment by ilyabirman** [comment]
Yes, thank you.

**Pull request merged**